packages/@webex/internal-plugin-dss/test/unit/spec/dss.ts

```diff
@@ -921,6 +1157,79 @@ describe('plugin-dss', () => {
         expect(Batcher.prototype.request.getCall(1).args).to.deep.equal(['id2']);
         expect(result).to.equal(response2);
       });
+
+      it('fails when mercury does not response but only the affected batch', async () => {
+        const requests = [
+          {
+            id: 'req-id-1',
+            resource: '/lookup/orgid/userOrgId/identities',
+            bodyParams: {lookupValues: ['id1', 'id2', 'id3']},
+          },
+        ];
+
+        const {
+          promises: [p1, p2, p3],
+        } = await testMakeBatchedRequests({
+          requests,
+          calls: [
+            {
+              method: 'lookup',
+              params: {id: 'id1', shouldBatch: true},
+            },
+            {
+              method: 'lookup',
+              params: {id: 'id2', shouldBatch: true},
+            },
+            {
+              method: 'lookup',
+              params: {id: 'id3', shouldBatch: true},
+            },
+          ],
+        });
+
+        // only 1 response
+        mercuryCallbacks['event:directory.lookup'](
```

Comment:
I wanted to make it more realistic. 

This proves timeout is still ticking even when a request finished.

Reply:
This mercury response doesn't finish any requests. If you meant to do 
createData('req-id-1', 0, true, 'lookupResult', {entitiesNotFound: ['id1', 'id3']})
instead of 
createData('id 1', 0, true, 'lookupResult', {entitiesNotFound: ['id1', 'id13']})
that would be an invalid mercury message.
if you meant to do 
createData('req-id-1', 0, false, 'lookupResult', {entitiesFound: ['id1', 'id3'], entities:['data1', 'data3']})
that would be a valid mercury message, though it wouldn't get back to the batcher, it's handled in the dss request code.